pkg/nvmeclient/nvme_client.go

```diff
@@ -521,11 +543,14 @@ func ConnectAllNVMEDevices(logPageEntries []*hostapi.NvmeDiscPageEntry, hostnqn
 				if perr.Status == CONN_ALREADY_CONNECTED {
 					continue
 				} else {
-					// This warn will occur every 5 sec in case the node is down.
-					// discovery service will still report this controller to connect to but we will fail to connect.
-					// we can't deduce that if the DS is down on that node we will fail to connect cause there might be a network partition
-					// on the discovery-service or the DS is down on that node but the IO controller is still accessible.
-					logrus.WithError(perr).Warnf("failed to connect IO controller. In case we have a node down, discovery provides log-page with it's connect info. we will probably fail to connect to it.")
+					ctrlID, err = connectNVMEDevicesWithRetry(request)
+					if errors.As(err, &perr) {
+						// This warn will occur every 5 sec in case the node is down.
```

Comment:
I know you're just copying the previous code but is it really every 5 secs with the new retry code?

Reply:
actually I'm not sure how this warning occurs every 5 sec w/o my new code as we don't return error and I don't see why `triggerReconnectToClusterCh` (with reconnectInterval of 5s) is triggered when node is down

Reply:
we discussed yesterday about it - 
1. we don't return error if we fail to connect nvme-device so the retry mechanism we have is irrelevant
2. no I don't know why test 271 fails